src/ynEIGEN/AssetRegistry.sol

```diff
@@ -286,8 +286,15 @@ interface IAssetRegistryEvents {
             revert LengthMismatch(assetsCount, stakedAssetBalances.length);
         }
 
+        IRedemptionAssetsVaultExt redemptionAssetsVault = strategyManager.redemptionAssetsVault();
```

Comment:
i think we should move all this logic to the EigenStrategyManager.sol

here it just introduces this additional dependency without another reason when we could simply return it as part of getStakedAssetBalances()